dmscripts/notify_suppliers_of_new_questions_answers.py

```diff
@@ -0,0 +1,159 @@
+EMAIL_FROM_ADDRESS = "do-not-reply@digitalmarketplace.service.gov.uk"
+EMAIL_FROM_NAME = "Digital Marketplace Team"
+
+
+logger = logging_helpers.configure_logger({'dmapiclient': logging.INFO})
+
+
+def get_live_briefs_with_new_questions_and_answers_between_two_dates(data_api_client, start_date, end_date):
+    briefs = data_api_client.find_briefs_iter(status='live', human=True, with_clarification_questions=True)
+
+    # return a list of briefs that contain clarification questions published between the start data and the end date
+    return [brief for brief in briefs if brief.get('clarificationQuestions') and any(
+        datetime.strptime(question['publishedAt'], DATETIME_FORMAT) >= start_date
+        and datetime.strptime(question['publishedAt'], DATETIME_FORMAT) <= end_date
+        for question in brief['clarificationQuestions']
+    )]
+
+
+def get_ids_of_suppliers_who_started_applying(data_api_client, brief):
+    responses = data_api_client.find_brief_responses(brief_id=brief["id"])
+    return [response["supplierId"] for response in responses["briefResponses"]]
+
+
+def get_ids_of_suppliers_who_asked_a_clarification_question(data_api_client, brief):
+    audit_events = data_api_client.find_audit_events(
+        audit_type=dmapiclient.audit.AuditTypes.send_clarification_question,
+        object_type='briefs',
+        object_id=brief['id']
+    )
+    return [audit_event['data']['supplierId'] for audit_event in audit_events['auditEvents']]
+
+
+def get_ids_of_interested_suppliers_for_briefs(data_api_client, briefs):
+    interested_suppliers = {}
+    for brief in briefs:
+        suppliers_who_applied = get_ids_of_suppliers_who_started_applying(data_api_client, brief)
+        suppliers_who_asked_a_question = get_ids_of_suppliers_who_asked_a_clarification_question(data_api_client, brief)
+        interested_suppliers[brief['id']] = list(set(suppliers_who_applied + suppliers_who_asked_a_question))
+
+    return invert_a_dictionary_so_supplier_id_is_key_and_brief_id_is_value(interested_suppliers)
+
+
+def invert_a_dictionary_so_supplier_id_is_key_and_brief_id_is_value(dictionary_to_invert):
+    inverted_dic = {}
+    for brief_id, list_of_supplier_ids in dictionary_to_invert.items():
+        for supplier_id in list_of_supplier_ids:
+            inverted_dic.setdefault(supplier_id, []).append(brief_id)
+
+    return inverted_dic
+
+
+def get_supplier_email_addresses_by_supplier_id(data_api_client, supplier_id):
+    response = data_api_client.find_users(supplier_id=supplier_id)
+    return [user['emailAddress'] for user in response['users']]
+
+
+def _get_link_domain(stage):
+    return env_helpers.get_api_endpoint_from_stage(stage, app='www')
+
+
+def create_context_for_supplier(stage, supplier_briefs):
+    return {
+        'briefs': [
+            {
+                'brief_title': brief['title'],
+                'brief_link': '{0}/{1}/opportunities/{2}'.format(
```

Comment:
Hmm, consistency is good, but so is doing something once rather than multiple times right next to each other.

So I'm tempted to say I'd change both to include the `utm_id` in the `create_context` rather than in the template.